src/UnigramWordPredictor.java

```diff
@@ -101,7 +146,38 @@ public void train(Scanner scanner) {
   public String predictNextWord(List<String> context) {
     // TODO: Return a predicted word given the words preceding it
     // Hint: only the last word in context should be looked at
-    return null;
+
+    // check if context is empty 
+    if (context.isEmpty()){
+      return null;
+    }
+
+    // grabbing the checkedWord
+    String checkedWord = context.get(context.size() - 1);
+
+
+
+    if(getNeighborMap().containsKey(checkedWord)){
```

Comment:
You can just use `neighborMap` instead of `getNeighborMap` because we're inside an instance method here.